### PR TITLE
Update hash for Kickstarter-Prelude and Kickstarter-ReactiveExtensions

### PR DESCRIPTION
**Pull request by justinswart**
- Updated maintainer details
- Added Kickstarter iOS app

### CODE REVIEW

**Comment by justinswart** [comment]
@swift-ci test

**Comment by justinswart** [comment]
@swift-ci test

**Comment by justinswart** [comment]
@clackary apologies for taking some time to get around to this. Let me know if I've done it correctly! 👍

**Comment by clackary** [comment]
@justinswart No worries! I should be able to get testing for it started in the next day or two. Our PR testing is down at the moment, and I'm working to get it back up.

File: projects.json [hunk 1]
```diff
@@ -828,8 +828,12 @@
...
     "branch": "master",
-    "maintainer": "stephen@stephencelis.com",
+    "maintainer": "justins@kickstarter.com",
     "compatibility": [
+      {
+        "version": "4.0",
+        "commit": "7b4eff0bae4f1abaf8ccc8370153f0945e0752ae"
+      },
       {
         "version": "3.0",
```
Comment (clackary) [inline projects.json]: Now that we have a 4.0 hashes, we can drop the 3.0 hashes. Would you mind updating this to remove the 3.0 configurations for Kickstarter-Prelude and Kickstarter-ReactiveExtensions? (That trailing comma after each 4.0 configuration also causes problems if it's not removed.)

Comment (justinswart) [reply projects.json]: Will do!

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by rudkx** [comment]
@justinswart It looks like this is hitting an issue because of a missing framework:

ld: framework not found Runes

**Comment by justinswart** [comment]
Apologies! Give me a bit of time to look into this, I had some issues running the pre-commit python script locally but have it running now so can try to debug.

**Comment by rudkx** [comment]
@justinswart No worries!

**Comment by justinswart** [comment]
🤔 I forgot that we need to run `make bootstrap` for our project to sync its dependencies and to also do some juggling with our app secrets file. Is it possible to run a custom command like this through the `projects.json` configuration file?

**Comment by clackary** [comment]
@justinswart It is not possible to run commands through projects.json. Might be possible to do it in a pre/post-build step in the Xcode scheme?

**Comment by clackary** [comment]
@swift-ci test

**Comment by justinswart** [comment]
@clackary just removing the stuff that isn't working for now until I can make some time to look into it a bit more

**Comment by justinswart** [comment]
I'd like to take a look at this again - is there an environment variable on your CI that I could check for to run `make bootstrap`? We don't run it on every build in Xcode so I would want to make it conditional, something like:

```bash
if [ -z "${SWIFTCI:-}" ]; then
  make bootstrap
fi
```

**Comment by clackary** [comment]
@justinswart I'm not sure about an environment variable like that. Hey @shahmishal, any idea if something like this exists?

**Comment by shahmishal** [comment]
I created a PR (https://github.com/apple/swift-source-compat-suite/pull/245) to add an environment variable to the source compatibility suite. Hopefully by adding the environment variable to the `project.py` it will build the same way as CI.

**Comment by shahmishal** [comment]
@justinswart Can you try using `SWIFT_SOURCE_COMPAT_SUITE` environment variable?

**Comment by justinswart** [comment]
Thanks @shahmishal I will try that!

**Comment by slavapestov** [comment]
It would be great to merge this soon - we want to remove Swift 3 support.

**Comment by justinswart** [comment]
@slavapestov apologies for the delay, @ifbarrera and I have some time set up to hand this over and get it wrapped up soonest!

**Comment by ifbarrera** [comment]
@swift-ci Please test source compatibility

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@ifbarrera Unfortunately, only users with commit access can trigger @swift-ci due to security reasons.

**Comment by justinswart** [comment]
We'd like to use this PR just to update the hashes for ReactiveExtensions and Prelude to 4.0 and then create a separate PR to add the app. We've done some work on that on our end but would like to spend a little more time on it and not hold this up.

Any ideas why this is failing with `14:42:17 ld: warning: Could not find auto-linked library 'swiftMetal'`?

**Comment by shahmishal** [comment]
New failure 

```
$ git -C /Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite-test-macOS/project_cache/Kickstarter-Prelude submodule update --init --recursive
Submodule 'Frameworks/Runes' (git://github.com/thoughtbot/Runes.git) registered for path 'Frameworks/Runes'
Cloning into '/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite-test-macOS/project_cache/Kickstarter-Prelude/Frameworks/Runes'...
fatal: unable to connect to github.com:
github.com[0: 192.30.255.112]: errno=Connection refused
github.com[1: 192.30.255.113]: errno=Connection refused

fatal: clone of 'git://github.com/thoughtbot/Runes.git' into submodule path '/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite-test-macOS/project_cache/Kickstarter-Prelude/Frameworks/Runes' failed
Failed to clone 'Frameworks/Runes'. Retry scheduled
Cloning into '/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite-test-macOS/project_cache/Kickstarter-Prelude/Frameworks/Runes'...
fatal: unable to connect to github.com:
github.com[0: 192.30.255.113]: errno=Connection refused
github.com[1: 192.30.255.112]: errno=Connection refused

fatal: clone of 'git://github.com/thoughtbot/Runes.git' into submodule path '/Users/buildnode/jenkins/workspace-private/swift-PR-source-compat-suite-test-macOS/project_cache/Kickstarter-Prelude/Frameworks/Runes' failed
Failed to clone 'Frameworks/Runes' a second time, aborting
```

**Comment by justinswart** [comment]
I thought we fixed the Runes issue 🤔 ok we'll check this out again on next week!

**Comment by justinswart** [comment]
@shahmishal In the log you posted it seems to say `github.com[0: 192.30.255.112]: errno=Connection refused`

Is that not a potential issue from the CI side?

**Comment by slavapestov** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@justinswart Found the issue with the submodule 

.gitmodules
```
[submodule "Frameworks/Runes"]
        path = Frameworks/Runes
        url = git://github.com/thoughtbot/Runes.git
```

Is it possible to switch this to be `url = https://github.com/thoughtbot/Runes.git`?

**Comment by justinswart** [comment]
@shahmishal sure - I can make that change 👍

**Comment by justinswart** [comment]
I've opened a PR on our repo for this change, also updated to Swift 4.2 (I'll update the hashes).

Are you sure about the URL, I think the one your originally posted was correct, tested in terminal:

```bash
$ git clone https://github.com/thoughtbot/Runes.git
Cloning into 'Runes'...
remote: Enumerating objects: 898, done.
remote: Total 898 (delta 0), reused 0 (delta 0), pack-reused 898
Receiving objects: 100% (898/898), 231.89 KiB | 827.00 KiB/s, done.
Resolving deltas: 100% (437/437), done.
$ git clone https://github.com/thoughtbot/Runes
fatal: destination path 'Runes' already exists and is not an empty directory.
```

**Comment by shahmishal** [comment]
If it works with `.git`, I am fine with that too. However, I looked at `.gitmodules` submodule from `Kickstarter-ReactiveExtensions` (https://github.com/kickstarter/Kickstarter-ReactiveExtensions/blob/master/.gitmodules)

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by justinswart** [comment]
If this fixes the PR then it's likely that the `git://` vs `https://` was the problem when we were adding the main app repo too.

If so then I will create a separate PR to add the app to the compatibility suite.

**Comment by clackary** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
```
PASS: Kickstarter-Prelude, 4.2, 8988b0, Prelude-iOS, generic/platform=iOS
PASS: Kickstarter-Prelude, 4.2, 8988b0, Prelude-UIKit-iOS, generic/platform=iOS
PASS: Kickstarter-ReactiveExtensions, 4.0, 6b7887, ReactiveExtensions-iOS, generic/platform=iOS
PASS: Kickstarter-ReactiveExtensions, 4.0, 6b7887, ReactiveExtensions-TestHelpers-iOS, generic/platform=iOS
```

**Comment by justinswart** [comment]
Nice! apologies once again for how long this took, we've just had a lot going on!

I'll create the app PR soon.

**Comment by shahmishal** [comment]
@justinswart @clackary Thanks!

**Pull request merged**